pkg/apis/camel/v1/common_types.go

```diff
@@ -36,8 +36,8 @@ const (
 )
 
 // BuildStrategy specifies how the Build should be executed.
-// It will trigger a Maven process that will take care of producing the expected Camel/Camel-Quarkus runtime.
-// +kubebuilder:validation:Enum=routine;pod
+// It will trigger a Maven daemon process that will take care of producing the expected Camel/Camel-Quarkus runtime. The Maven daemon will be run in a container. 
```

Comment:
I'd suggest to keep the existing documentation, that's generic enough to cover the new strategy as well.